test/unit/foreman/renderer/template_snapshot_service.rb

```diff
@@ -0,0 +1,31 @@
+class TemplateSnapshotService
+  SNAPSHOTS_DIRECTORY = Rails.root.join('test', 'unit', 'foreman', 'renderer', 'snapshots')
+
+  def self.templates
+    new.templates
+  end
+
+  def self.host
+    new.host
+  end
+
+  def templates
+    template_names = ProvisioningTemplatesList.seeded_templates.select {|item| item[:snapshot_test] }.pluck(:name)
+    template_names.map { |template_name| Template.find_by!(name: template_name) }
+  end
+
+  def host
```

Comment:
Wouldn't using factory bot here be simpler than manually setting all these attributes?